### PR TITLE
Adding protobuffer helper

### PR DESCRIPTION
**Pull request by Honcharov12**
Add protobuf_helper.py module which allows to store some functions that fills Protocol Buffer objects in one separate place.

### CODE REVIEW

**Comment by Honcharov12** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/5741/

**Comment by jeanleonov** [comment]
Just to make it easier to read the PR:
this branch is based on https://github.com/AppScale/appscale/pull/2931 and https://github.com/AppScale/appscale/pull/2913, so those PRs should be merged first.

File: SearchService/protobuf_helper.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+from google.appengine.datastore.document_pb import FieldValue
+
+
+def fill_protobuf_doc(gae_doc, solr_doc, index):
+  """ Fill new document from a query result.
+
+  Args:
+    solr_doc: A dictionary of SOLR document attributes.
+    gae_doc: A search_service_pb.SearchResult.
+    index: Index we queried for.
```
Comment (jeanleonov) [inline SearchService/protobuf_helper.py]: Update docstring to match arguments order.

File: SearchService/protobuf_helper.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  Raises:
+    UnknownFieldException: If there are no some fields in indexes schema.
```
Comment (jeanleonov) [inline SearchService/protobuf_helper.py]: `If solr_doc contains field(s) which is missing in index schema.`

File: SearchService/protobuf_helper.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      raise UnknownFieldException('Unable to find type for {}_{}'.format(index.name, field_name))
+    fill_protobuf_field(new_value, solr_doc[key], field_type)
+
+
+def fill_protobuf_field(gae_field, solr_field, ftype):
+  """ Fills search_service_pb.SearchResult field with a value.
+
+  Args:
+    gae_field: A search_service_pb.SearchResult field.
+    solr_field: Field value in SOLR.
```
Comment (jeanleonov) [inline SearchService/protobuf_helper.py]: `solr_field: A dict representing Solr document field.`

File: SearchService/protobuf_helper.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    geo = gae_field.mutable_geo()
+    lat, lng = solr_field.split(',')
+    geo.set_lat(float(lat))
+    geo.set_lng(float(lng))
+    gae_field.set_type(FieldValue.GEO)
+  elif ftype.startswith(Field.TEXT_):
+    gae_field.set_string_value(solr_field)
+    gae_field.set_type(FieldValue.TEXT)
+  else:
+    raise UnknownFieldException("Default field {} not found!".format(ftype))
```
Comment (jeanleonov) [inline SearchService/protobuf_helper.py]: It sounds like it should be `UnknownFieldTypeException("Solr document contains a field of unknown type: {}".format(ftype))`.
I'd suggest to rename exception class.

**Pull request closed without merging**